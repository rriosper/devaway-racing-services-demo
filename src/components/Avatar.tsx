import styled, { css } from '@xstyled/styled-components';
import { variant } from '@xstyled/system';

type AvatarProps = {
  size?: 'sm' | 'md' | 'lg';
};

const Avatar = styled.img<AvatarProps>`
  border-radius: 50%;
  background-color: foregroundBorder;
  ${variant({
    prop: 'size',
    default: 'md',
    variants: {
      sm: css`
        width: md;
        height: md;
      `,
      md: css`
        width: lg;
        height: lg;
      `,
      lg: css`
        width: xl;
        height: xl;
      `,
    },
  })}
`;

export default Avatar;
