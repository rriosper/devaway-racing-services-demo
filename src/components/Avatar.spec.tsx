import React from 'react';
import { render } from '@testing-library/react';

import Avatar from './Avatar';

test('Avatar', () => {
  const { asFragment } = render(<Avatar />);
  expect(asFragment).toMatchSnapshot();
});
