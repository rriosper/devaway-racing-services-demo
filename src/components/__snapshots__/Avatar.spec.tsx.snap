// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar 1`] = `[Function]`;
